### PR TITLE
Resolved enrollmentDate missing issue in export data and experiment details state time logs

### LINKED ISSUE #2099: EnrolmentStart date not stored for state time logs on updating the experiment state to enrolling
**Version where bug was found:**
e.g "6.1.0"

**Describe the bug**
Enrolment Start date not stored for state time logs on updating the experiment state to enrolling for a few QA experiments. Due to this, even the experiment CSV data export shows wrong/NA date for this.

**To Reproduce**
1. Go to the experiment in staging: https://upgrade.qa-cli.com/home/detail/bdb0b762-04ce-45db-a929-aaa9fc85e264 name "TC-RB-ISP-test".
2. Click on Time Logs.
3. It would show blank data, though experiment is in enrolling state.

**Expected behavior**
We should properly store the enrolment start date.

**Screenshot**
Screenshot 2024-11-07 at 6 30 08 PM

### PR DESCRIPTION
**Pull request by ppratikcr7**
This resolves #2099

### CODE REVIEW

**Review by zackcl — DISMISSED** [review]

**Comment by zackcl** [comment]
I can confirm the Enrollment Time Log is added when an experiment with the "Enrolling" status is created with POST `/api/experiments`.

**Comment by ppratikcr7** [comment]
> I can confirm the Enrollment Time Log is added when an experiment with the "Enrolling" status is created with POST `/api/experiments`.

@zackcl This PR need to handle update experiment api. This was pointed out by Vivek. Will push another commit for it.

**Comment by ppratikcr7** [comment]
> I can confirm the Enrollment Time Log is added when an experiment with the "Enrolling" status is created with POST `/api/experiments`.

@zackcl You can review it now.

File: backend/packages/Upgrade/src/api/services/ExperimentService.ts [hunk 1]
```diff
@@ -1134,6 +1138,16 @@ export class ExperimentService {
       });
   }
 
+  async updateStateTimeLogs(experimentDoc: Experiment, transactionalEntityManager: EntityManager) {
+    const stateTimeLogDoc = new StateTimeLog();
+    stateTimeLogDoc.id = uuid();
+    stateTimeLogDoc.fromState = EXPERIMENT_STATE.INACTIVE;
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/services/ExperimentService.ts:1144]: You are hard-coding `fromState` here but it is getting called from `updateExperimentInDB`. What if the experiment is in a different state than `Inactive`.

**Review by VivekFitkariwala — DISMISSED** [review]

**Review by zackcl — APPROVED** [review]

**Review by VivekFitkariwala — APPROVED** [review]

**Review by VivekFitkariwala — APPROVED** [review]

**Pull request merged**